great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -223,6 +221,52 @@ def _get_dialect_type_module(dialect):
     return dialect
 
 
+_PERSISTED_CONNECTION_DIALECTS = (GXSqlDialect.SQLITE, GXSqlDialect.MSSQL)
+
+
+def _dialect_requires_persisted_connection(
+    connection_string: str | None = None,
+    credentials: dict | None = None,
+    url: str | None = None,
+) -> bool:
+    """Determine if the dialect needs a persisted connection.
+
+    dialect_name isn't available yet since the engine isn't yet created when we call this method
+    so we determine the dialect from the creds/url/params:
+
+    Args:
+        connection_string:
+        credentials:
+        url:
+
+    Returns:
+
+    """
+    if sum(bool(x) for x in [connection_string, credentials, url is not None]) != 1:
```

Comment:
I think it can be, but it will either error here if multiple params are passed or later when trying to parse it. I added some additional tests. Let me know if that works!